app/models/user.rb

```diff
@@ -22,16 +21,51 @@ def email_required?
   end
 
   def check_if_email_or_phone_entered?
-    if email.blank? && phone_number.blank?
+    if self.email.blank? && self.phone_number.blank?
       errors.add(:email, "must have an email or a phone number!")
     end
+    if !self.use_email? && !self.use_call? && !self.use_text?
+      errors.add(:use_email, "must select contact preferences!")
```

Comment:
Since all of these add the same error method, should this just be a private message that accepts a field? Additionally, should this error message be in our `I18n` file?